src/common/common.cpp

```diff
@@ -218,28 +218,31 @@ int run_message_loop() {
   return static_cast<int>(msg.wParam);
 }
 
+std::optional<std::wstring> get_last_error_message(const DWORD dw) {
+  std::optional<std::wstring> message;
+  try {
+    const auto msg = std::system_category().message(dw);
+    message.emplace(begin(msg), end(msg));
+  }
+  catch(...) {
+    
+  }
+  return message;
+}
+
 void show_last_error_message(LPCWSTR lpszFunction, DWORD dw) {
-  // Retrieve the system error message for the error code
-  LPWSTR lpMsgBuf = NULL;
-  if (FormatMessageW(FORMAT_MESSAGE_ALLOCATE_BUFFER |
-                     FORMAT_MESSAGE_FROM_SYSTEM |
-                     FORMAT_MESSAGE_IGNORE_INSERTS,
-                     NULL,
-                     dw,
-                     MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
-                     lpMsgBuf,
-                     0, NULL) > 0) {
-    // Display the error message and exit the process
-    LPWSTR lpDisplayBuf = (LPWSTR)LocalAlloc(LMEM_ZEROINIT, (lstrlenW(lpMsgBuf) + lstrlenW(lpszFunction) + 40) * sizeof(WCHAR));
-    if (lpDisplayBuf != NULL) {
-      StringCchPrintfW(lpDisplayBuf,
-        LocalSize(lpDisplayBuf) / sizeof(WCHAR),
-        L"%s failed with error %d: %s",
-        lpszFunction, dw, lpMsgBuf);
-      MessageBoxW(NULL, (LPCTSTR)lpDisplayBuf, L"Error", MB_OK);
-      LocalFree(lpDisplayBuf);
-    }
-    LocalFree(lpMsgBuf);
+  const auto system_message = get_last_error_message(dw);
+  if(!system_message.has_value()) {
+    return;
+  }
+  LPWSTR lpDisplayBuf = (LPWSTR)LocalAlloc(LMEM_ZEROINIT, (system_message->size() + lstrlenW(lpszFunction) + 40) * sizeof(WCHAR));
+  if (lpDisplayBuf != NULL) {
+    StringCchPrintfW(lpDisplayBuf,
+      LocalSize(lpDisplayBuf) / sizeof(WCHAR),
+      L"%s failed with error %d: %s",
```

Comment:
Since you are making changes to this code, let's take the time to make the string localizable.
Same for "Error" at line 244.